server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -117,10 +117,11 @@ public static Map<String, Object> shape2Map(Shape shape) {
             for (Shape collShape : shapeCollection) {
                 geometries.add(shape2Map(collShape));
             }
-            return Map.of(
+            // Transform Geometry Collection back to MultiPoint/MultiPolygon if possible.
+            return GeoJSONUtils.validateGeoJson(Map.of(
```

Comment:
Integration test (I will replace it later with unit tests) passes without it, ie comparison of literals, insert + select works.

`test_check_equality_with_itself`  fails without this change but not sure if test code path is really relevant to any user facing scenario:
compare object literal with normalized version which is `type.implicitCast(GeoJSONUtils.map2Shape(geoJson));`

I will look for a scenario where we can hit such combination.

Reply:
>I will look for a scenario where we can hit such combination 

I was not able to find a user facing scenario where we need this change. 
This change was targeting a failing test originally suggested by @mfussenegger but I think we cannot call `shape2Map` with non-normalized values - the only user facing usage is WKT string which already does that transformation. 

Reverted in 
https://github.com/crate/crate/pull/17012/commits/974d87d2cbb73822a55020f071e76aec5be32ac9